agent/php_internal_instrument.c

```diff
@@ -796,6 +800,35 @@ static void nr_php_instrument_datastore_operation_call(
   }
 }
 
+/* 
+ * Like nr_php_instrument_datastore_operation_call() but does not
+ * call the underlying internal function. This allows multiple
+ * datastore metrics to be creating during a single underlying
+ * PHP call.
+ */
+static void nr_php_instrument_datastore_operation(
+    nr_datastore_t datastore,
+    const char* operation,
+    nr_datastore_instance_t* instance,
+    bool instance_only) {
+  nr_segment_t* segment = NULL;
+  nr_segment_datastore_params_t params = {
+      .datastore = {
+          .type = datastore,
+      },
+      .operation = nr_strdup(operation),
+      .instance  = instance,
+      .instance_only = instance_only,
+      .callbacks = {
+          .backtrace = nr_php_backtrace_callback,
+      },
+  };
+
+  segment = nr_segment_start(NRPRG(txn), NULL, NULL);
```

Comment:
Do we need to create a datastore segment?  Since it's not associated with any datastore calls, should it create the segment?  Would it be sufficient to simply send an instance metric?

Reply:
I was thinking the same. I did some digging into the code and unfortunately it looks like separating instance metric creation from other pieces of `nr_segment_datastore_end` is rather involved (if not impossible). But I'm curious of @ZNeumann reasoning behind it.

Reply:
I initially did try to create this metric without also doing the datastore segment. I agree that it would be very difficult to do, and -- considering this seems to be the only current use case for such a paradigm -- decided against it

Reply:
very good idea. I have made the change here [3ed5bc5](https://github.com/newrelic/newrelic-php-agent/pull/958/commits/3ed5bc54bb0d90bc792ed32cd650cfb87c7de972) and cleaned out the now-unused code here [ffb0711](https://github.com/newrelic/newrelic-php-agent/pull/958/commits/ffb0711d8e562d4e1ece30c0dd9b557bc49d04ef)